arangod/Pregel/IncomingCache.cpp

```diff
@@ -103,7 +103,7 @@ void InCache<M>::storeMessage(PregelShard shard, std::string_view vertexId,
 // ================== ArrayIncomingCache ==================
 
 template<typename M>
-ArrayInCache<M>::ArrayInCache(std::set<PregelShard> localShards,
+ArrayInCache<M>::ArrayInCache(containers::FlatHashSet<PregelShard> localShards,
                               MessageFormat<M> const* format)
     : InCache<M>(format), _localShards(localShards) {
   // one mutex per shard, we will see how this scales
```

Comment:
In the following loop, each `PregelShard` is copied. This is unnecessary. The range-based for loop should use a const reference to PregelShard.